hotham-editor/src/main.rs

```diff
@@ -0,0 +1,362 @@
+            }
+
+            Event::MainEventsCleared => {
+                let framebuffer_index = lazy_vulkan.render_begin();
+                send_swapchain_image_index(&mut stream, &mut buf, framebuffer_index);
+                get_render_complete(&mut stream, &mut buf);
+                let texture_id = textures[framebuffer_index as usize].id;
+                lazy_renderer.render(
+                    lazy_vulkan.context(),
+                    framebuffer_index,
+                    &[DrawCall::new(
+                        0,
+                        indices.len() as _,
+                        texture_id,
+                        lazy_vulkan::Workflow::Main,
+                    )],
+                );
+
+                let semaphore = semaphores[framebuffer_index as usize];
+                lazy_vulkan.render_end(
+                    framebuffer_index,
+                    &[semaphore, lazy_vulkan.rendering_complete_semaphore],
+                );
+            }
+            Event::WindowEvent {
+                event: WindowEvent::Resized(size),
+                ..
+            } => {
+                if !winit_initializing {
+                    let new_render_surface = lazy_vulkan.resized(size.width, size.height);
+                    lazy_renderer.update_surface(new_render_surface, &lazy_vulkan.context().device);
+                }
+            }
+            _ => (),
+        }
+    });
+
+    // I guess we better do this or else the Dreaded Validation Layers will complain
+    unsafe {
+        lazy_renderer.cleanup(&lazy_vulkan.context().device);
+    }
+}
+
+fn send_semaphore_handles(
+    stream: &mut UnixStream,
+    semaphore_handles: Vec<*mut std::ffi::c_void>,
+    buf: &mut [u8; 1024],
+) {
+    stream.read(buf).unwrap();
+    let value = buf[0];
+    debug!("Read {value}");
+
+    debug!("Sending handles: {semaphore_handles:?}");
+    let write = stream.write(bytes_of_slice(&semaphore_handles)).unwrap();
+    debug!("Wrote {write} bytes");
+}
+
+unsafe fn create_semaphores(
+    context: &lazy_vulkan::vulkan_context::VulkanContext,
+    image_count: u32,
+) -> (Vec<vk::Semaphore>, Vec<vk::HANDLE>) {
+    let device = &context.device;
+    let external_semaphore =
+        ash::extensions::khr::ExternalSemaphoreWin32::new(&context.instance, &context.device);
+    let handle_type = vk::ExternalSemaphoreHandleTypeFlags::OPAQUE_WIN32_KMT;
+    (0..image_count)
+        .map(|_| {
+            let mut external_semaphore_info =
+                vk::ExportSemaphoreCreateInfo::builder().handle_types(handle_type);
+            let semaphore = device
+                .create_semaphore(
+                    &vk::SemaphoreCreateInfo::builder().push_next(&mut external_semaphore_info),
+                    None,
+                )
+                .unwrap();
+
+            let handle = external_semaphore
+                .get_semaphore_win32_handle(
+                    &vk::SemaphoreGetWin32HandleInfoKHR::builder()
```

Comment:
😨 only windows?

Reply:
Nope! This can be done on each platform, but each platform has its own unique way of doing it. Focussing on Windows for the first pass.